contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/modeling/polynomial/LinearBoundaryConsistencySolver.java

```diff
@@ -132,8 +125,11 @@ private void solve() {
                   "LinearBoundaryConsistencySolver %s failed. Domain for %s is empty: [%s, %s]".formatted(
                       getName(this).orElseThrow(), D.variable, D.lowerBound, D.upperBound));
             }
-            // TODO: Make this more efficient by not adding constraints that are already in the queue.
-            remainingConstraints.addAll(neighboringConstraints.get(D.variable));
+            for (DirectionalConstraint constraintToAdd : neighboringConstraints.get(D.variable)) {
+              if (!remainingConstraints.contains(constraintToAdd)) {
```

Comment:
Makes me wonder if a LinkedHashSet would be appropriate here

Reply:
That seems like it would work. With the change to clampedIntegrate away from using this solver, optimizing the solver is less important. Using a set instead of a list would simplify the code, though.